tests/Controller/AdminProductVariantAjaxTest.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Sylius\Tests\Controller;
+
+use ApiTestCase\JsonApiTestCase;
+use Symfony\Component\BrowserKit\Cookie;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\UsernamePasswordToken;
+
+class AdminProductVariantAjaxTest extends JsonApiTestCase
+{
+    /**
+     * @test
+     */
+    public function it_denies_access_to_product_variants_for_not_authenticated_user()
+    {
+        $this->client->request('GET', '/admin/ajax/product-variants/search-all');
+
+        $response = $this->client->getResponse();
+
+        $this->assertEquals($response->getStatusCode(), Response::HTTP_UNAUTHORIZED);
+    }
+
+    /**
+     * @test
+     */
+    public function it_returns_only_specified_part_of_all_product_variants()
+    {
+        $this->authenticateAdminUser();
+
+        $this->client->request('GET', '/admin/ajax/product-variants/search-all');
```

Comment:
What about other behaviours? Those with added functionalities (max result and page parameters?)